src/components/BulkUploaderFilePicker.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {
+  ChangeEventHandler,
+  DragEventHandler,
+  MouseEventHandler,
+  useRef,
+  useState,
+} from 'react';
+import './BulkUploaderFilePicker.css';
+
+export const BulkUploaderFilePicker = () => {
+  const ACCEPTED_MIME_TYPES = ['text/csv'];
+  const ACCEPTED_FILE_EXTENSIONS = ['csv'];
+  const IDLE_MESSAGE = 'Drag and drop your CSV file here.';
+
+  const [message, setMessage] = useState(IDLE_MESSAGE);
+
+  const [dragging, setDragging] = useState(false);
+  const [uploading, setUploading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const formRef = useRef<HTMLFormElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  /**
+   * Provided a list of files
+   * from either a native file picker
+   * or the Drag and Drop API,
+   * performs pre-upload validation
+   * and then uploads the files.
+   *
+   * @param  {FileList | null} files
+   */
+  const handleFiles = (files: FileList | null) => {
+    if (!files) {
+      return;
+    }
+
+    if (files.length > 1) {
+      setMessage('You can only upload one file at a time.');
+      setError(true);
+      return;
+    }
+
+    // TODO: Validate file (type, size, ?)
+
+    // Remove focus from the button.
+    // We wait until after validation
+    // so focus is not wrenched away.
+    buttonRef.current?.blur();
+
+    setMessage('Uploading your file…');
+    setUploading(true);
+
+    // TODO: Begin uploading
```

Comment:
I think that's an inevitable advancement, though I thought it might wait for when we have a second upload example so we better understand the generic vs. specific surface area? Still, if you think it's something that should happen now and/or is obvious enough how to extract, go for it! (I doubt I will before you need to implement. Apologies for that, truly.)